sdk/spring-sdk/src/main/java/kalix/springsdk/annotations/Subscribe.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.springsdk.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+public @interface Subscribe {
+
+  @Target(ElementType.METHOD)
+  @Retention(RetentionPolicy.RUNTIME)
+  @interface ValueEntity {
+    String entityType();
+  }
```

Comment:
I would like to experiment with this new naming. If we feel that is more intuitive, we should change the proto annotations to reflect it. 

Basically, when annotating a class with this, we end-up with: 

```java
@Subscribe.ValueEntity(entityType = "counter")
  public Effect<Done> changes(@RequestBody CounterState counterState) {
    logger.info("Counter subscriber: counter id '{}' is '{}'", counterState.id, counterState.value);
    return effects().reply(Done.instance);
  }
```

I believe this reads better. "Subscribe to the ValueEntity with entityType defined as 'counter'"

If we would make it similar to the current protobuf annotations, it will be something like:
`@EventingIn.ValueEntity(source = "counter")`

Reply:
The idea is to have the following:

```java
@Subscribe.ValueEntity(entityType = "counter")

@Subscribe.EventSourcedEntity(entityType = "counter")

@Subscribe.Topic(name = "counter")
```
and then 

```java
@Publish.Topic(name = "counter")
```

Reply:
Maybe, when subscribing to entities from another service we will want to have another kind of annotations.

Reply:
Yes, enforcing at runtime/startup is what I meant, I don't think the type system can help at compile time.

Can't we enforce the value entity / es enttity type of consumption anyway by looking at what type the entity subscribed to is?

Reply:
We can enforce at compile time, but then we need `@Subscribe.ValueEntity` and `@Subscribe.EventSourcedEntity`. 
Or by introducing a common trait, like `PersistentEntity` (too exclude replicated ones) and having a single annotation: `@Subscribe.Entity`

However, I think we should have two distinct ones so we can block the usage of  `@Subscribe.EventSourcedEntity` at type level. 

Another reason for distinct annotations is that it will make clear that we can't subscribe to any kind of entities, but only for VE and ES.

Reply:
I tried it out. It works.